js/src/sdk/models/Entity.ts

```diff
@@ -125,14 +139,34 @@ export class Entity {
     }
   }
 
-  async getConnection({ app, connectedAccountId }: ConnectionParams) {
+  /**
+   * Retrieves the required parameters for a specific authentication scheme of an app in the Composio platform.
+   *
+   * This method allows clients to fetch the necessary parameters for a specific authentication scheme of an app by providing its unique key and the authentication scheme.
+   *
+   * @param {ConnectionParams} data The data for the request, including the app's unique key and the authentication scheme.
+   * @returns {Promise<RequiredParamsResponse>} A promise that resolves to the required parameters for the authentication scheme.
+   * @throws {ComposioError} If the request fails.
+   */
+  async getConnection({ app, appName, connectedAccountId }: ConnectionParams) {
     TELEMETRY_LOGGER.manualTelemetry(TELEMETRY_EVENTS.SDK_METHOD_INVOKED, {
       method: "getConnection",
       file: this.fileName,
-      params: { app, connectedAccountId },
+      params: { app, appName, connectedAccountId },
     });
     try {
-      ZConnectionParams.parse({ app, connectedAccountId });
+      const finalApp = appName || app;
+      ZConnectionParams.parse({ app: finalApp, connectedAccountId });
+
+      if (!finalApp && !connectedAccountId) {
```

Comment:
Consider adding validation for the entityId parameter at the schema level using Zod. This would catch invalid values earlier and provide better error messages. For example:

```typescript
const ZEntityParams = z.object({
  entityId: z.string().min(1, "Entity ID cannot be empty"),
  connectedAccountId: z.string().optional()
});
```

This would provide consistent validation across all methods that use these parameters.